boot/boot_serial/src/boot_serial.c

```diff
@@ -589,9 +589,8 @@ boot_serial_output(void)
     bs_hdr->nh_group = htons(bs_hdr->nh_group);
 
 #ifdef __ZEPHYR__
-    crc =  crc16((uint8_t *)bs_hdr, sizeof(*bs_hdr), CRC_CITT_POLYMINAL,
-                 CRC16_INITIAL_CRC, false);
-    crc =  crc16(data, len, CRC_CITT_POLYMINAL, crc, true);
+    crc =  crc16_itu_t(CRC16_INITIAL_CRC, (uint8_t *)bs_hdr, sizeof(*bs_hdr));
+    crc =  crc16_itu_t(crc, data, len);
```

Comment:
It looks that both the `crc16_itu_t` and `crc16_ccitt` calls are exactly the same now. Maybe we could get rid of the `#ifdef __ZEPHYR__` from the common code by providing definitions that would redirect, for Zephyr, calls from `crc16_ccitt` to `crc16_itu_t`, or some inline functions.

Reply:
>It looks that both the crc16_itu_t and crc16_ccitt calls are exactly the same now.

In Zephyr they are not, but in other operating systems (NuttX, FreeRTOS, Mynewt) I have no idea.